doc/sphinx/proofs/writing-proofs/proof-mode.rst

```diff
@@ -1,66 +1,165 @@
 .. _proofhandling:
 
--------------------
- Proof handling
--------------------
+----------
+Proof mode
+----------
 
-In Coq’s proof editing mode all top-level commands documented in
-Chapter :ref:`vernacularcommands` remain available and the user has access to specialized
-commands dealing with proof development pragmas documented in this
-section. They can also use some other specialized commands called
-*tactics*. They are the very tools allowing the user to deal with
-logical reasoning. They are documented in Chapter :ref:`tactics`.
+:gdef:`Proof mode <proof mode>` is used to prove theorems.
+Coq enters proof mode when you begin a proof,
+such as with the :cmd:`Theorem` command.  It exits proof mode when
+you complete a proof, such as with the :cmd:`Qed` command.  Tactics,
+which are available only in proof mode, incrementally transform incomplete
+proofs to eventually generate a complete proof.
 
-Coq user interfaces usually have a way of marking whether the user has
-switched to proof editing mode. For instance, in coqtop the prompt ``Coq <``   is changed into
-:n:`@ident <`   where :token:`ident` is the declared name of the theorem currently edited.
+When you run Coq interactively, such as through CoqIDE, Proof General or
+coqtop, Coq shows the current proof state--the incomplete proof--as you
+enter tactics.  This information isn't shown when you run Coq in batch
+mode with `coqc`.
 
-At each stage of a proof development, one has a list of goals to
-prove. Initially, the list consists only in the theorem itself. After
-having applied some tactics, the list of goals contains the subgoals
-generated by the tactics.
+Proof State
+-----------
 
-To each subgoal is associated a number of hypotheses called the *local context*
-of the goal. Initially, the local context contains the local variables and
-hypotheses of the current section (see Section :ref:`gallina-assumptions`) and
-the local variables and hypotheses of the theorem statement. It is enriched by
-the use of certain tactics (see e.g. :tacn:`intro`).
+.. todo PR: my example is pretty bad--can you suggest a better one?
+   Also need to position/integrate this better
 
-When a proof is completed, the message ``Proof completed`` is displayed.
-One can then register this proof as a defined constant in the
-environment. Because there exists a correspondence between proofs and
-terms of λ-calculus, known as the *Curry-Howard isomorphism*
-:cite:`How80,Bar81,Gir89,H89`, Coq stores proofs as terms of |Cic|. Those
-terms are called *proof terms*.
+The :gdef:`proof state`, also referred to as the :gdef:`proof context`, consists of one
+or more unproven goals.
+Each goal has a :gdef:`conclusion` (the statement that is to be proven)
+and may have zero or more named facts that can be used in proving the
+conclusion (in addition to definitions, proven theorems, etc.).
```

Comment:
This is precisely what I (and others I think) call the "proof context". It comes in addition to the "global context".

Reply:
How often do we need to refer to "proof context" as something different from "proof state" and "goal context"?  "local context" is used 58 times while "global context" is used 5 times in the doc.  The latter probably should be used more often.

Reply:
OK then "global environment" is actually more common. Does it mean that we should uniformize toward "global environment" and "local context"?

Reply:
I'm thinking "global context" and "local context" but I'm open to suggestions.